tests/connection/AConnectionTest.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace Bolt\tests\connection;
+
+use Bolt\Bolt;
+use Bolt\connection\IConnection;
+use Bolt\connection\Socket;
+use Bolt\connection\StreamSocket;
+use Bolt\error\ConnectionTimeoutException;
+use Bolt\error\MessageException;
+use Bolt\helpers\Auth;
+use Bolt\protocol\V4;
+use PHPUnit\Framework\TestCase;
+
+final class AConnectionTest extends TestCase
+{
+    public function provideConnections(): array
+    {
+        return [
+            [StreamSocket::class],
+            [Socket::class],
+        ];
+    }
+
+    /**
+     * @dataProvider provideConnections
+     */
+    public function testMillisecondTimeout(string $alias): void
+    {
+        $socket = $this->getConnection($alias);
+        $protocol = (new Bolt($socket))->build();
+        $socket->setTimeout(1.5);
+        $protocol->init(Auth::basic($GLOBALS['NEO_USER'], $GLOBALS['NEO_PASS']));
+
+        $time = microtime(true);
+        try {
+            $protocol->run('FOREACH ( i IN range(1,10000) | 
+  MERGE (d:Day {day: i})
+)');
+            $this->fail('No timeout error triggered');
+        } catch (ConnectionTimeoutException $e) {
+            $newTime = microtime(true);
+
+            $this->assertEqualsWithDelta(1.5, $newTime - $time, 0.2);
+        }
+    }
+
+    /**
+     * @dataProvider provideConnections
+     */
+    public function testSecondsTimeout(string $alias): void
+    {
+        $socket = $this->getConnection($alias);
+        $protocol = (new Bolt($socket))->build();
+        $protocol->init(Auth::basic($GLOBALS['NEO_USER'], $GLOBALS['NEO_PASS']));
+
+        $time = microtime(true);
+        try {
+            $protocol->run('FOREACH ( i IN range(1,10000) | 
+  MERGE (d:Day {day: i})
+)');
+            $this->fail('No timeout error triggered');
+        } catch (ConnectionTimeoutException $e) {
+            $newTime = microtime(true);
+
+            $this->assertEqualsWithDelta(1.0, $newTime - $time, 0.2);
+        }
+    }
+
+    /**
+     * @dataProvider provideConnections
+     */
+    public function testTimeoutRecoverAndReset(string $alias): void
+    {
+        $socket = $this->getConnection($alias);
```

Comment:
this is interesting. you define `$socket` variable but don't use it. instead you make `$streamSocket` in next line. How it should be?

Reply:
This is why a second pair of eyes is absolutely necessary :sweat_smile:, I completely meant to use socket, not streamsocket.

fixed in commit d978783